### PR TITLE
SUP-14294: removed validation of parentId from CuePoint and move it t…

### PR DESCRIPTION
**Pull request by Adirvkaltura**
…o annotation

### CODE REVIEW

File: plugins/cue_points/annotation/lib/api/KalturaAnnotation.php [hunk 1]
```diff
@@ -147,4 +147,37 @@ public function validateForUpdate($sourceObject, $propertiesToSkip = array())
...
+
+    /*
+ * @param string $cuePointId
+ * @throw KalturaAPIException - when parent annotation doesn't belong to the same entry, or parent annotation
+ * doesn't belong to the same entry
+ */
+    public function validateParentId($cuePointId = null)
+    {
+
+        if (!$this->parentId)
```
Comment (MosheMaorKaltura) [inline plugins/cue_points/annotation/lib/api/KalturaAnnotation.php]: Logic is reversed, should check only if there is value

File: plugins/cue_points/annotation/lib/api/KalturaAnnotation.php [hunk 2]
```diff
@@ -147,4 +147,37 @@ public function validateForUpdate($sourceObject, $propertiesToSkip = array())
 			
 		return parent::validateForUpdate($sourceObject, $propertiesToSkip);
 	}
+
+    /*
+ * @param string $cuePointId
+ * @throw KalturaAPIException - when parent annotation doesn't belong to the same entry, or parent annotation
```
Comment (MosheMaorKaltura) [inline plugins/cue_points/annotation/lib/api/KalturaAnnotation.php]: Fix remark

Comment (Adirvkaltura) [reply plugins/cue_points/annotation/lib/api/KalturaAnnotation.php]: fixed

File: plugins/cue_points/annotation/lib/api/KalturaAnnotation.php (hunk 1 above)
Comment (Adirvkaltura) [reply plugins/cue_points/annotation/lib/api/KalturaAnnotation.php]: fixed

**Comment by MosheMaorKaltura** [comment]
reviewed 
Please make sure that you test - 
Adding Annotation and Answer cuepoint with parent and without.

**Comment by MosheMaorKaltura** [comment]
Reviewed - please merge

**Pull request merged**